### ISSUE TITLE
Design discussion: extending GROBID to books / edited volumes with per-chapter metadata

### ISSUE THREAD

**Issue by cboulanger** [kermitt2/grobid#1538]
**Summary:** GROBID currently assumes one PDF = one work with one header and one author set, which doesn't fit books — especially edited volumes, where each chapter has its own title, author(s), and bibliographic identity under a shared volume-level context. This issue surveys existing but unwired scaffolding for book handling (`MonographParser`'s `unit` label) and opens two design questions for discussion: (1) where chapter splitting should happen in the processing pipeline, and (2) what TEI output shape should represent a multi-chapter, multi-author volume.

## Motivation

GROBID currently treats an input PDF as a single scholarly work: one `teiHeader`, one author set, one bibliographic identity. This fits journal articles and theses well, but breaks down for **edited volumes**, where each chapter has its own title, author(s), and (arguably) its own bibliographic identity, while still sharing volume-level metadata (editors, ISBN, publisher, table of contents, etc.).

With recent improvements to pdfalto's handling of large PDFs, processing full-length books is now more tractable, which makes this a good time to open a discussion on how GROBID's pipeline and TEI output should be extended to support them — before committing to an implementation.

This issue is meant to start that discussion, not to propose a final design. It lays out what already exists in the codebase toward this goal, and two open questions that need input from maintainers/the community before implementation work begins.

## What already exists

A search of the codebase turned up an existing, but currently **disconnected**, building block:

- **`MonographParser`** (`grobid-core/.../engines/MonographParser.java`) performs a high-level zone segmentation of a monograph (book, thesis, proceedings volume, etc.) into 16 labels: `cover`, `titlePage`, `publisherPage`, `summary`, `biography`, `advertising`, `toc`, `tof`, `preface`, `dedication`, **`unit`** ("chapter or standalone article"), `reference`, `annex`, `index`, `glossary`, `back`, `other`.
- The `unit` label is the closest existing concept to "chapter" — but it only marks a chapter's *extent* within the page geometry. `MonographParser.processing()` returns a segmented `Document` and stops there: it does **not** run header/fulltext extraction per unit, and has **no TEI serialization** of its own.
- The model is **not wired into production**: its initialization is commented out in `EngineParsers.java` (`//tryInit(() -> monographParser = getMonographParser(), "monograph");`), it has no REST endpoint in `grobid-service`, and it's reachable only via the trainer/batch CLI.
- There is **no existing precedent** in GROBID for emitting multi-work TEI output. `teiCorpus` appears only as inherited TEI-schema plumbing (module exclusion lists, compiled ODD/RNG/DTD) and in unrelated patent test fixtures — never as something GROBID itself generates. There's also no existing "multi-article-in-one-PDF" (e.g. journal issue, proceedings) handling to reuse as precedent.

**Conclusion:** any book/chapter-metadata feature is largely greenfield work, but it can plausibly build on top of (or replace) `MonographParser` and its `unit` label rather than starting from nothing.

## Open question 1: where does chapter splitting happen in the pipeline?

The existing `segmentation` model works well for single-work documents (header/body/biblio/footnote/etc. zones) and is structurally well-suited to operate **per chapter**, since a chapter is structurally similar to a standalone article. Two candidate architectures:

**Option A — Dedicated book-segmentation stage.**
Revive and extend `MonographParser` (or a new model built on the same `unit` concept) as a first pass that splits the volume into chapter-extent zones. Each resulting chapter is then handed to the existing `segmentation` → `header` → `fulltext` pipeline unchanged.

Pros:

- Keeps `segmentation`'s training data and label set focused on single-work structure, no regressions to existing article-processing accuracy.
- Reuses the entire existing per-article pipeline as-is for each chapter.
- Matches the existing (if unfinished) `unit` label's intent — natural continuation of prior design intent in the codebase.

Cons:

- Needs a new/extended training corpus for the book-segmentation stage specifically (chapter-boundary annotation at the volume level), separate from `segmentation`'s corpus.
- Introduces a second model into the cascade, with its own maintenance/accuracy burden and failure mode (a bad chapter split propagates errors into every downstream chapter).

**Option B — Extend the existing `segmentation` model.**
Add chapter-boundary detection as new label(s) directly in the existing `segmentation` model's label set, alongside `header`/`body`/`biblio`/etc.

Pros:

- No second model to introduce, train, or maintain.

Cons:

- Complicates `segmentation`'s label set and training data with book-only concerns, for a model whose accuracy on the far more common single-article case is presumably a priority to protect.
- Less clean separation of concerns — `segmentation` would need to reason simultaneously about volume-level chapter boundaries and within-chapter zones.
- Training data implications: existing `segmentation` training TEI corpus would need chapter-boundary annotations added throughout, mixing article-level and book-level ground truth in the same files.

Both options carry different training-data consequences, noted above per-option, so we haven't split that out as a separate question.

We don't have a strong recommendation between these yet — **input from maintainers on which fits GROBID's architecture and roadmap better is exactly what this issue is meant to surface.**

## Open question 2: what should the output TEI structure be?

Once chapters are identified and individually processed, how should the final TEI represent a volume with multiple, separately-authored chapters?

**Option A — Single `<TEI>` with `<group>`.**

```xml
<TEI xml:id="collection_001" xml:lang="en">
  <teiHeader> <!-- volume-level: editors, ISBN, publisher --> </teiHeader>
  <text>
    <front> <!-- preface, ToC --> </front>
    <body>
      <group>
        <text xml:id="chapter_01"> <!-- chapter 1, own byline/author markup within --> </text>
        <text xml:id="chapter_02"> <!-- chapter 2 --> </text>
      </group>
    </body>
    <back> <!-- index, contributor list --> </back>
  </text>
</TEI>
```

Pros:

- One coherent document identity for the whole volume; front/back matter attaches naturally to a single document.

Cons:

- Per-chapter author/title metadata has to live in in-text markup (e.g. `<byline>`, nested `<div type="chapter">`) rather than a full independent `teiHeader`, since `<text>` inside `<group>` has no header of its own.

**Option B — `<teiCorpus>` wrapping per-chapter `<TEI>`.**

```xml
<teiCorpus>
  <teiHeader> <!-- volume-level: editors, publication info --> </teiHeader>
  <TEI xml:id="chapter_01"> <!-- full independent TEI, own teiHeader with chapter author --> </TEI>
  <TEI xml:id="chapter_02"> <!-- ... --> </TEI>
</teiCorpus>
```

Pros:

- Each chapter gets a fully independent `teiHeader` — cleanest separation of per-chapter bibliographic identity (author, title, own biblStruct), and closest to how GROBID already treats one-TEI-per-document elsewhere in the codebase.

Cons:

- Volume-level front/back matter (ToC, preface, index) has no natural home — `teiCorpus` only defines a shared header plus a sequence of `TEI` elements, not front/back matter of its own.

Both are valid TEI; the trade-off is essentially "one document with rich internal structure" vs. "a corpus of fully independent documents with a shared wrapper header." We'd like input on which shape better serves downstream consumers of GROBID's TEI output (indexing pipelines, citation extraction, etc.) before settling on one.

## Next steps

I'd like to invite feedback from maintainers and the community on:

1. Pipeline architecture (Option A vs. B in the first question) — including whether reviving/extending `MonographParser` is the right foundation.
2. TEI output shape (Option A vs. B in the second question) — including whether there's a downstream-tooling reason to prefer one over the other that isn't visible from the codebase alone.

**Comment by cboulanger** [comment]
Another way of having independent chapter metadata with Option 2a cound be using `xml:id` to connect chapters with separate `biblStruct` sections in the document's `teiHeader`, I don't know if that's legal per the TEI spec.
